presto-parquet/src/main/java/com/facebook/presto/parquet/reader/AbstractColumnReader.java

```diff
@@ -130,10 +138,13 @@ public ColumnChunk readNext()
                 readNextPage();
             }
             int valuesToRead = Math.min(remainingValueCountInPage, nextBatchSize - valueCount);
+            if (valuesToRead == 0) {
+                // When we break here, we could end up with valueCount < nextBatchSize, but that is OK.
+                break;
+            }
             readValues(blockBuilder, valuesToRead, field.getType(), definitionLevels, repetitionLevels);
             valueCount += valuesToRead;
         }
-        checkArgument(valueCount == nextBatchSize, "valueCount %s not equals to batchSize %s", valueCount, nextBatchSize);
```

Comment:
This is mainly because we skip row reading with page index. We could end up with a situation that valueCount < nextBatchSize.  The earlier assumption won't be true anymore.